File: December - 03/SakthiDharan01_bridge_crossing_challenge.java
import java.util.*;
import java.util.regex.*;

public class SakthiDharan01_bridge_crossing_challenge {
    private static List<Integer> parse(String input) {
        Matcher m = Pattern.compile("-?\\d+").matcher(input);
        List<Integer> nums = new ArrayList<>();
        while (m.find()) {
            nums.add(Integer.parseInt(m.group()));
        }
        if (nums.size() > 1 && nums.get(0) == nums.size() - 1) {
            return nums.subList(1, nums.size());
        }
        return nums;
    }

    private static boolean canReach(List<Integer> arr) {
        int far = 0;
        for (int i = 0; i < arr.size(); i++) {
            if (i > far) return false;
            far = Math.max(far, i + arr.get(i));
            if (far >= arr.size() - 1) return true;
        }
        return far >= arr.size() - 1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        while (sc.hasNextLine()) sb.append(sc.nextLine()).append(' ');
        List<Integer> arr = parse(sb.toString());
        if (arr.isEmpty()) return;
        System.out.println(Boolean.toString(canReach(arr)).toLowerCase());
        sc.close();
    }
}

File: December - 03/SakthiDharan01_bridge_crossing_challenge.py
import sys
import re

def parse_array(data: str):
    nums = list(map(int, re.findall(r"-?\d+", data)))
    if not nums:
        return []
    if len(nums) > 1 and nums[0] == len(nums) - 1:
        return nums[1:]
    return nums

def can_reach(arr):
    far = 0
    for i, v in enumerate(arr):
        if i > far:
            return False
        far = max(far, i + v)
        if far >= len(arr) - 1:
            return True
    return far >= len(arr) - 1

def solve():
    data = sys.stdin.read()
    arr = parse_array(data)
    if not arr:
        return
    print(str(can_reach(arr)).lower())

if __name__ == "__main__":
    solve()
